cache_store_redis/lib/cache_store_redis.rb

```diff
@@ -1,117 +1,159 @@
+# frozen_string_literal: true
+
+require 'thread'
 require 'cache_store_redis/version'
 require 'redis'
 require 'securerandom'
-require 'connection_pool'
 
-#This class is used to implement a redis cache store.
-#This class is used for interacting with a redis based cache store.
+# This class is used to implement a redis cache store.
 class RedisCacheStore
-
-  def self.pool(config)
-    @pool ||= ConnectionPool.new(size: pool_size, timeout: pool_timeout) do
-      if config == nil
-        Redis.new
-      else
-        Redis.new(config)
-      end
+  def initialize(namespace = nil, config = nil)
+    unless RUBY_PLATFORM == 'java'
+      require 'oj'
     end
-  end
 
-  def self.pool_size
-    @pool_size ||= Integer(ENV['CACHE_STORE_POOL_SIZE'] || 20)
-  end
+    @namespace = namespace
+    @config = config
+    @queue = Queue.new
 
-  def self.pool_timeout
-    @pool_size ||= Integer(ENV['CACHE_STORE_POOL_TIMEOUT'] || 1)
+    @connections_created = 0
+    @connections_in_use = 0
+    @mutex = Mutex.new
+    @enable_stats = false
   end
 
-  def initialize(namespace = nil, config = nil)
+  def enable_stats=(value)
+    @enable_stats = value
+  end
 
-    if RUBY_PLATFORM != 'java'
-      require 'oj'
+  def increment_created_stat
+    @mutex.synchronize do
+      @connections_created += 1
     end
+  end
 
-    @namespace = namespace
-    @config = config
+  def increment_using_stat
+    @mutex.synchronize do
+      @connections_in_use += 1
+    end
   end
 
-  def with_client
-    if self.class.pool_size > 0
-      self.class.pool(@config).with(timeout: self.class.pool_timeout) do |client|
-        raise 'nil redis client returned from pool' if client.nil?
-        yield client
-      end
-    else
-      # allow single client to be used when pool size is zero
-      @client ||= Redis.new(@config)
-      yield @client
+  def decrement_using_stat
+    @mutex.synchronize do
+      @connections_in_use -= 1
     end
   end
 
-  #This method is called to configure the connection to the cache store.
-  def configure(host = 'localhost', port = 6379, db = 'default', password = nil, driver: nil, url: nil)
-    if url != nil
+  # This method is called to configure the connection to the cache store.
+  def configure(host = 'localhost',
+                port = 6379,
+                db = 'default',
+                password = nil,
+                driver: nil,
+                url: nil,
+                connect_timeout: 0.5,
+                read_timeout: 1,
+                write_timeout: 0.5)
+    if !url.nil?
       @config = {}
       @config[:url] = url
       @config[:db] = db
     else
-      @config = { :host => host, :port => port, :db => db }
+      @config = { host: host, port: port, db: db }
     end
-    if password != nil
-      @config[:password] = password
+
+    @config[:password] = password unless password .nil?
```

Comment:
`password.nil?`

Reply:
good spot not sure what's happened with that formatting